terraform/cloud-platform-components/prometheus.tf

```diff
@@ -0,0 +1,74 @@
+resource "helm_repository" "coreos" {
+  name = "coreos"
+  url  = "https://s3-eu-west-1.amazonaws.com/coreos-charts/stable/"
+}
+
+resource "kubernetes_storage_class" "prometheus" {
+  metadata {
+    name = "prometheus-storage"
+  }
+
+  storage_provisioner = "kubernetes.io/aws-ebs"
+
+  parameters {
+    type = "gp2"
+  }
+}
+
+resource "helm_release" "prometheus_operator" {
+  name          = "prometheus-operator"
+  chart         = "coreos/prometheus-operator"
+  namespace     = "monitoring-terraform"
+  recreate_pods = "true"
+
+  depends_on = [
+    "null_resource.deploy",
+    "helm_repository.coreos",
+  ]
+}
+
+resource "helm_release" "kube_prometheus" {
+  name          = "kube-prometheus"
+  chart         = "coreos/kube-prometheus"
+  namespace     = "monitoring-terraform"
+  recreate_pods = "true"
+
+  values = [
+    "${file("../../helm-charts/prometheus-operator/kube-prometheus/values.yaml")}",
+  ]
+
+  set {
+    name  = "grafana.extraVars[0].value"
```

Comment:
This feels too brittle (what if we re-order elements?) - also the same for the slack configs below.
Should we instead treat `values.yaml` as a template file and use terraform interpolation instead of overriding using `set {}` ?

Reply:
I think this is fair. The set command is how Helm recommends doing interpolation but we may as well use Terraform. 

I have made a rather big change in 652b1b3 and 3ee0687.

Helm docs:
https://docs.helm.sh/chart_best_practices/values/#consider-how-users-will-use-your-values